tests/test_cases/test_plusargs/plusargs.py

```diff
@@ -45,5 +46,8 @@ def plusargs_test(dut):
 
     for name in cocotb.plusargs:
         print("COCOTB:", name, cocotb.plusargs[name])
+    
+    if cocotb.plusargs['foo'] != 'bar':
+        raise TestFailure("plusargs 'foo' value '%s' does not match expected 'bar'" % cocotb.plusargs['foo'])
```

Comment:
`% cocotb.plusargs['foo']` -> `% (cocotb.plusargs['foo'], ) or use `.format()`

Reply:
You could also extend the test to check if argc gets passed as expected by checking `len(cocotb.plusargs)`

Reply:
I'm not sure checking `len(cocotb.plusargs)` will actually work correctly between multiple simulators.
Aldec Active-HDL / Riviera-PRO often uses `+access +w` arguments passed in the simulation startup command that are also parsed by cocotb, whereas I'm not sure that other simulators do this.